test/unit/Parser.spec.ts

```diff
@@ -36,8 +36,9 @@ describe('Parser', () => {
     return new Promise((resolve, reject) => {
       wait(packets.length)
         .then((parsedPackets) => {
-          for (let i = 0; i <= packets.length; i += 1) {
+          for (let i = 0; i < packets.length; i += 1) {
             expect(packets[i]).to.deep.equal(parsedPackets[i]);
+            expect(packets[i].type).to.equal(parsedPackets[i].type);
```

Comment:
Isn't this redundant? The deep equal check above should also check `type`.

Reply:
Without it the test would have passed (while having the bug) - haven't looked much into it, but probably because `type` is a getter and so a function, not a plain field.

Reply:
Ohhh that's right, that makes sense. That's fine.